### ISSUE TITLE
Allow shortening the command line using a pathing JAR on Mac OSX

### ISSUE THREAD

**Issue by rpalcolea** [gradle/gradle#11907]
https://github.com/gradle/gradle/pull/10544 and https://github.com/gradle/gradle/pull/11620 introduced support for pathing jar on Windows//Unix/OSX. However, Mac OS is not working as expected.

Given this sample project: https://github.com/rpalcolea/test-gradle-pathing-jar

When running 

```
./gradlew -g /tmp/really/really/really/long/prefix/for/each/dependency/j3awDqlz1NftNYlWdFtbR0jLlOE2kYrs8UGkijDPKLsJu8AUms9WpXAJUMMMhTijn3MA5LixQRb66Hm6c6ZM87DSxwANfxDrRQNz/test-gradle-path-jar bootRun -s
```

You will get the argument list too long exception:

```
Caused by: java.io.IOException: Cannot run program "/Users/rperezalcolea/.sdkman/candidates/java/8.0.181-zulu/bin/java" (in directory "/Users/rperezalcolea/Projects/github/rpalcolea/test-gradle-path-jar"): error=7, Argument list too long
        at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
```

Doing some debugging I found that my command line length is `306934` and the max is set to `2097152`  inhttps://github.com/gradle/gradle/blob/d4571f51c877dd0d9f946957b3b4e4c527f2ce26/subprojects/core/src/main/java/org/gradle/process/internal/util/LongCommandLineDetectionUtil.java#L26 so this condition is never `true` fir https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/process/internal/JavaExecHandleBuilder.java#L328 so the pathing jar is never created.

I manipulated the value via debugger to see if things work if it passes the limit…
a gradle-javaexec JAR is created `/private/var/folders/8x/bhy9sgq52ydc4lwkyzlxlwv80000gn/T/gradle-javaexec-classpath6090803244857022210.jar`
And this JAR only contains a MANIFEST with the path for all dependencies (as expected).

Looking at the default `ARG_MAX` in `limits.h`, I found that is `262144` which differs from the default value on Linux distributions

It would be great if `LongCommandLineDetectionUtil` supported the limit on OSX.

## Environment info

```
------------------------------------------------------------
Gradle 6.1-rc-3
------------------------------------------------------------

Build time:   2020-01-13 16:23:24 UTC
Revision:     52aebd9acc0e395deba9a17db22f6cffe7d61cf1

Kotlin:       1.3.61
Groovy:       2.5.8
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          1.8.0_181 (Azul Systems, Inc. 25.181-b02)
OS:           Mac OS X 10.14 x86_64
```

**Comment by lacasseio** [comment]
Thanks for the contribution, it is really appreciated and was of great quality. We merged it and will be present in Gradle 6.2.

**Comment by rpalcolea** [comment]
Thanks for looking at this @lacasseio, having this in gradle core is phenomenal, so thanks again for the nice work on this feature!
